runtime/src/main/java/org/corfudb/runtime/view/workflows/AddNode.java

```diff
@@ -0,0 +1,68 @@
+package org.corfudb.runtime.view.workflows;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.protocols.wireprotocol.orchestrator.CreateWorkflowResponse;
+import org.corfudb.runtime.CorfuRuntime;
+import org.corfudb.runtime.clients.ManagementClient;
+import org.corfudb.runtime.view.Layout;
+
+import javax.annotation.Nonnull;
+import java.time.Duration;
+import java.util.List;
+import java.util.UUID;
+
+/**
+ *
+ * A workflow request that makes an orchestrator call to add a new node to
+ * the cluster.
+ *
+ * Created by Maithem on 1/19/18.
+ */
+@Slf4j
+public class AddNode extends WorkflowRequest {
+
+    private final String endpointToAdd;
+
+    public AddNode(@Nonnull String endpointToAdd, @Nonnull CorfuRuntime runtime,
+                   @Nonnull int retry, @Nonnull Duration timeout,
+                   @Nonnull Duration pollPeriod) {
+        this.endpointToAdd = endpointToAdd;
+        this.runtime = runtime;
+        this.retry = retry;
+        this.timeout = timeout;
+        this.pollPeriod = pollPeriod;
+    }
+
+    @Override
+    protected UUID sendRequest(@Nonnull Layout layout) {
+        // Select the current tail node and send an add node request to the orchestrator
+        CreateWorkflowResponse resp = getOrchestrator(layout).addNodeRequest(endpointToAdd);
+        log.info("sendRequest: requested to add {} on orchestrator {}:{}, layout {}",
+                endpointToAdd, getOrchestrator(layout).getRouter().getHost(),
+                getOrchestrator(layout).getRouter().getPort(), layout);
+        return resp.getWorkflowId();
+    }
+
+    @Override
+    protected ManagementClient getOrchestrator(@Nonnull Layout layout) {
+        // Select the current tail node and send an add node request to the orchestrator
+        List<String> logServers = layout.getSegments().get(0).getStripes().get(0)
+                .getLogServers();
+        String orchestratorEndpoint = logServers.get(logServers.size() - 1);
```

Comment:
I think orchestrators are going to reside where the management servers are and Management servers are more likely to be with Layout Servers.
So you should find an orchestrator endpoint there. (LayoutServers - UnresponsiveServers)

Reply:
Done.